src/wandbot/chat/rag.py

```diff
@@ -59,22 +60,23 @@ def __init__(
         )
         self.response_synthesizer = ResponseSynthesizer()
 
+    @weave.op()
     def __call__(
         self, question: str, chat_history: List[Tuple[str, str]] | None = None
-    ):
+    ) -> RAGPipelineOutput:
         if chat_history is None:
             chat_history = []
 
         with get_openai_callback() as query_enhancer_cb, Timer() as query_enhancer_tb:
-            enhanced_query = self.query_enhancer.chain.invoke(
+            enhanced_query = self.query_enhancer(
```

Comment:
is the requirement of not doing `.chain.invoke` a new thing in LECL?

Reply:
I see you have added `__call__` methods.

Reply:
haha, yes, this allows the use of a `weave.op` decorator, making the weave integration easier.